### ISSUE TITLE
DumpStyle.HTML Dump Option

### ISSUE THREAD

**Issue by lawrencek76** [thomasgalliker/ObjectDumper#39]
I was looking at adding another dump format for html.  Is that something you would be interested in bringing into ObjectDumper?

I'm looking to create simple html output that would be useful in debugging web apps / web app error pages.  html formatted diagnostic email messages.

**Comment by thomasgalliker** [comment]
Can you make an example of how such an html message would look like?

**Comment by lawrencek76** [comment]
The primary output would be something like this for the example in the readme.  
```HTML
    <div class="obj-dump">
        <div class="obj-class-info">
            <span class="obj-label">Class Type:</span>
            <span
                class="obj-type-list-objectdumpersample-netfx-person">List&lt;ObjectDumperSample.Netfx.Person&gt;</span>
        </div>
        <table class="obj-type-list-objectdumpersample-netfx-person">
            <thead>
                <tr>
                    <th class="obj-type-string">Name</th>
                    <th class="obj-type-int">Age</th>
                </tr>
            </thead>
            <tbody>
                <tr>
                    <td class="obj-type-string">John</td>
                    <td class="obj-type-int">20</td>
                </tr>
                <tr>
                    <td class="obj-type-string">Thomas</td>
                    <td class="obj-type-int">30</td>
                </tr>
            <tbody>
        </table>
    </div>
````
Styling would be primarily left to the caller to add although it might be nice to have options for creating a full html document with some basic styling that generates something like this.

![Capture](https://user-images.githubusercontent.com/1901007/60674256-ba7afe00-9e47-11e9-9b30-35ea46ab38cc.PNG)


[objDumperHtmlSample.txt](https://github.com/thomasgalliker/ObjectDumper/files/3359744/objDumperHtmlSample.txt)

Of course there are lot of details to flesh out on this but I think that shows the general idea.

**Comment by thomasgalliker** [comment]
Yes why not integrating something like that. Kind of a report-style output. The thing is just: There will be people adking for different styles and formatting of the html output. Of course, we could build a highly flexible formatting logic but I think that would be a bit over the top, what do you think?

**Comment by lawrencek76** [comment]
Yea simple at least at first. Adding classes so most styling can be done on the users end.  Not trying to go crazy with options and flexibility.

**Comment by thomasgalliker** [comment]
I'm sorry for not coming back to this in time. I saw you're working on branch lawrence-Html. Is this still on? Does it already work?

**Comment by lawrencek76** [comment]
Yea still on,  Just vacation is over, back to work.  It is working in the simple case f of object with properties and looking at how to deal with the more interesting case of collections of objects and making a html table out of it with nesting and all that.  

The branch also has some refactor of the "core" logic into the base class, that will probably be split off and done as a separate pr.
